[[36m2023-09-05 06:31:37,270[39m][[34m__main__[39m][[32mINFO[39m] - Instantiating trainer <lightning.pytorch.trainer.Trainer>
[[36m2023-09-05 06:31:37,353[39m][[34m__main__[39m][[32mINFO[39m] - Logging hyperparameters!
Running validation before training
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
In lf_datamodule.py, self.rho_name pol
Trainer already configured with model summary callbacks: [<class 'lightning.pytorch.callbacks.rich_model_summary.RichModelSummary'>]. Skipping setting a default `ModelSummary` callback.
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430: PossibleUserWarning: The dataloader, val_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m      Validate metric      [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         val/loss          [39m│[35m    15.259321212768555     [39m│
└───────────────────────────┴───────────────────────────┘
[37mValidation[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:00 • 0:00:00[39m [37m87.78it/s
[?25h[[36m2023-09-05 06:31:41,546[39m][[34m__main__[39m][[32mINFO[39m] - Starting training!
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
┏━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━┓
┃[1m    [22m┃[1m Name               [22m┃[1m Type                  [22m┃[1m Params [22m┃
┡━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━┩
│ 0  │ net                │ SoftplusSSM           │  1.4 K │
│ 1  │ net.W              │ CustomLinearLayer     │     16 │
│ 2  │ net.P              │ CustomOrthogonalLayer │    256 │
│ 3  │ net.g              │ MLP                   │    544 │
│ 4  │ net.g.linear1      │ Linear                │    272 │
│ 5  │ net.g.linear2      │ Linear                │    272 │
│ 6  │ net.f              │ MLP                   │    544 │
│ 7  │ net.f.linear1      │ Linear                │    272 │
│ 8  │ net.f.linear2      │ Linear                │    272 │
│ 9  │ encoder            │ Linear                │     32 │
│ 10 │ encoder.U          │ Linear                │     32 │
│ 11 │ encoder.activation │ Identity              │      0 │
│ 12 │ decoder            │ Linear                │     17 │
│ 13 │ decoder.U          │ Linear                │     17 │
│ 14 │ decoder.activation │ Identity              │      0 │
│ 15 │ train_loss         │ MeanMetric            │      0 │
│ 16 │ val_loss           │ MeanMetric            │      0 │
│ 17 │ test_loss          │ MeanMetric            │      0 │
└────┴────────────────────┴───────────────────────┴────────┘
[1mTrainable params[22m: 1.4 K
[1mNon-trainable params[22m: 0
[1mTotal params[22m: 1.4 K
[1mTotal estimated model params size (MB)[22m: 0
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430:
PossibleUserWarning: The dataloader, train_dataloader, does not have many workers which may be a bottleneck. Consider
increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the
`DataLoader` init to improve performance.
  rank_zero_warn(
[37mEpoch 0/999[39m [90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m2/250[39m [37m0:00:00 • 0:00:20[39m [37m12.81it/s[39m [37mv_num: 5wzk train/loss: 14.447 









Epoch 1/999 [35m━╸[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m13/250[39m [37m0:00:01 • 0:00:20[39m [37m12.18it/s[39m [37mv_num: 5wzk train/loss: 0.948       
                                                                                    [37mval/loss: 0.876                     
Metric val/loss improved. New best score: 0.876


Epoch 2/999 [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸[90m━━━━[39m [37m219/250[39m [37m0:00:17 • 0:00:03[39m [37m12.81it/s[39m [37mv_num: 5wzk train/loss: 0.268       
                                                                                    [37mval/loss: 0.651                     
                                                                                    [37mval/loss: 0.155                     
[?25h[[36m2023-09-05 06:32:44,610[39m][[34m__main__[39m][[32mINFO[39m] - Starting testing!
Restoring states from the checkpoint path at /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130/checkpoints/epoch_002.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
Loaded model weights from the checkpoint at /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130/checkpoints/epoch_002.ckpt
LF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m      Validate metric      [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         val/loss          [39m│[35m     0.154877707362175     [39m│
└───────────────────────────┴───────────────────────────┘
[37mValidation[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:00 • 0:00:00[39m [37m68.22it/s
[?25h
Restoring states from the checkpoint path at /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130/checkpoints/epoch_002.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
Loaded model weights from the checkpoint at /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130/checkpoints/epoch_002.ckpt
/home/shida/anaconda3/envs/lh/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:430: PossibleUserWarning: The dataloader, test_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 32 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
[?25hLF_generate done
In lf datagen, input shape (153600, 100, 1)
In lf datagen, output shape (153600, 100, 1)
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃[1m        Test metric        [22m┃[1m       DataLoader 0        [22m┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│[36m         test/loss         [39m│[35m    0.15620969235897064    [39m│
└───────────────────────────┴───────────────────────────┘
[37mTesting[39m [35m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[39m [37m25/25[39m [37m0:00:00 • 0:00:00[39m [37m67.65it/s
[?25h[[36m2023-09-05 06:32:48,696[39m][[34m__main__[39m][[32mINFO[39m] - Best ckpt path: /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130/checkpoints/epoch_002.ckpt
[[36m2023-09-05 06:32:48,699[39m][[34msrc.utils.utils[39m][[32mINFO[39m] - Output dir: /home/shida/Figure_1/logs/LF_tanh_softplusssm_pol/runs/20230905/063130
[[36m2023-09-05 06:32:48,699[39m][[34msrc.utils.utils[39m][[32mINFO[39m] - Closing wandb!